feder/cases/models.py

```diff
@@ -99,6 +114,35 @@ def update_email(self):
             pk=self.pk, domain=self.monitoring.domain.name
         )
 
+    @property
+    def application_letter(self):
+        return (
+            apps.get_model("letters", "Letter")
+            .objects.filter(record__case=self, author_user_id__isnull=False)
+            .order_by("created")
+            .first()
+        )
+
+    def get_confirmation_received(self):
```

Comment:
Nie rozumiem koncepcji tego gettera, skoro takie informacje trzymasz zdenormalizowane na modelu. Albo denormalizujesz (czy jest to konieczne? przypuszczam wydajność, ale do rozważenia jest czy możemy rozwiązać to np. optymalnymi indeksami), albo pobierasz dane rzeczywiste. Gdyby to był setter, aby odświeżyć właściwość modelu to bym mógł łatwiej zrozumieć.

Reply:
Może faktycznie jako setter będzie to czytelniejsze. Denormalizuję celowo, dla wygody dostępu do tych danych i ponieważ jest problem z wydajnością zapytań dot. raportu.